Formula/terminator.rb

```diff
@@ -0,0 +1,42 @@
+class Terminator < Formula
+  include Language::Python::Virtualenv
+
+  desc "Multiple GNOME terminals in one window"
+  homepage "https://gnome-terminator.org"
+  url "https://github.com/gnome-terminator/terminator/archive/refs/tags/v2.1.1.tar.gz"
+  sha256 "ee1907bc9bfe03244f6d8074b214ef1638a964b38e21ca2ad4cca993d0c1d31e"
+  license "GPL-2.0-only"
+
+  depends_on "dbus"
+  depends_on "pygobject3"
+  depends_on "python@3.9"
+  depends_on "vte3"
+
+  resource "dbus-python" do
+    url "https://files.pythonhosted.org/packages/62/7e/d4fb56a1695fa65da0c8d3071855fa5408447b913c58c01933c2f81a269a/dbus-python-1.2.16.tar.gz"
+    sha256 "11238f1d86c995d8aed2e22f04a1e3779f0d70e587caffeab4857f3c662ed5a4"
+  end
+
+  resource "psutil" do
+    url "https://files.pythonhosted.org/packages/e1/b0/7276de53321c12981717490516b7e612364f2cb372ee8901bd4a66a000d7/psutil-5.8.0.tar.gz"
+    sha256 "0c9ccb99ab76025f2f0bbecf341d4656e9c1351db8cc8a03ccd62e318ab4b5c6"
+  end
+
+  def install
+    # virtualenv_install_with_resources
+    venv = virtualenv_create(libexec, "python3")
+    %w[dbus-python psutil].each do |r|
+      venv.pip_install resource(r)
+    end
+    venv.pip_install_and_link buildpath
+
+    system "#{Formula["python@3.9"].opt_bin}/python3", "setup.py", "build"
+    system "#{Formula["python@3.9"].opt_bin}/python3", "setup.py", "install", "--prefix=#{prefix}"
+  end
+
+  test do
+    ENV["LANG"] = "en_US.UTF-8"
+    o = shell_output("DYLD_FALLBACK_LIBRARY_PATH=#{HOMEBREW_PREFIX}/lib #{bin}/terminator --version 2>&1")
+    assert_match version.to_s, o
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests.

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Simulating a pty here might help. See formulae that do `require "pty"` for examples.

Reply:
The only thing that terminator does is just open a terminal window.  The existing test is definitely imperfect, but I was scratching my head as to how to test this further.  The only other thing  I can think of is to spawn the window itself, and then kill it if it was successful.  If it's successful spawning a window, that exercises most of the code.

Reply:
We could try that. Is there a flag we can pass for the window spawning to generate a log or something similar? Then we can also test that it generates a log file successfully (and maybe check the log for some strings to make sure it's not broken).

Reply:
That's done with the new commit.